src/main/java/org/traccar/storage/DatabaseStorage.java

```diff
@@ -291,6 +303,18 @@ private String formatCondition(Condition genericCondition, boolean appendWhere)
                 result.append(getStorageName(Device.class));
                 if (condition.getDeviceId() > 0) {
                     result.append(" WHERE id = :deviceId");
+                } else if (condition.getDeviceIds() != null) {
+                    result.append(" WHERE id IN ( ");
+                    int paramMultiple = (condition.getDeviceIds().size() / 10) + 1;
+                    for (int i = 0; i < paramMultiple; i++) {
+                        for (int j = 0; j < 10; j++) {
+                            result.append(String.format(":deviceId%s", j + (10 * i)));
+                            if (i != (paramMultiple - 1) || j != 9) {
+                                result.append(", ");
+                            }
+                        }
+                    }
+                    result.append(" )");
```

Comment:
This definitely doesn't look right. If a user has thousands of devices, they shouldn't be included here one by one.